src/tests/system/tests/test_sssctl.py

```diff
@@ -269,3 +273,681 @@ def test_sssctl__check_misplaced_option(client: Client):
 
     pattern = re.compile(r".Attribute 'services' is not allowed in section .*")
     assert pattern.search(result.stdout), "Wrong error message was returned"
+
+
+@pytest.mark.tools
+@pytest.mark.topology(KnownTopology.Client)
+def test_sssctl__check_invalid_option_name_in_snippet(client: Client):
+    """
+    :title: sssctl config-check detects invalid option name in snippet
+    :setup:
+        1. Create new conf snippet with invalid option name
+    :steps:
+        1. Call sssctl config-check
+        2. Check error message
+    :expectedresults:
+        1. config-check detects an error in config snippet
+        2. Error message is properly set
+    :customerscenario: False
+    """
+    client.sssd.common.local()
+    client.fs.write("/etc/sssd/conf.d/01_snippet.conf", "[domain/local]\ninvalid_option = True", mode="600")
```

Comment:
Might be useful to add an interface for this like `client.sssd.dom_snippet` ? Not necessary for this PR though.

Reply:
Good point, I will put it to my backlog